### PR TITLE
Multiple discovery

### PR DESCRIPTION
**Pull request by jistefan**
Added new provider dicovery test, corrected amqp_credentials handling in provider.py.

{{pytest: cfme/tests/infrastructure/test_discover_multiple_providers.py -v --use-provider complete}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 6ebb8581ac3eee2c30ee4b1e2647ce8672fafc24:

`utils/providers.py`:
- :bangbang: Line 44:5: **E123** *closing bracket does not match indentation of opening bracket's line*
- :bangbang: Line 51:5: **E123** *closing bracket does not match indentation of opening bracket's line*
- :bangbang: Line 57:5: **E123** *closing bracket does not match indentation of opening bracket's line*
- :bangbang: Line 62:5: **E123** *closing bracket does not match indentation of opening bracket's line*
- :bangbang: Line 393:13: **E123** *closing bracket does not match indentation of opening bracket's line*
- :bangbang: Line 398:13: **E123** *closing bracket does not match indentation of opening bracket's line*
- :bangbang: Line 403:13: **E123** *closing bracket does not match indentation of opening bracket's line*
- :bangbang: Line 408:13: **E123** *closing bracket does not match indentation of opening bracket's line*
- :bangbang: Line 409:9: **E123** *closing bracket does not match indentation of opening bracket's line*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c6560d868636f256f84ff9650cfa51a89d23dec6:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9598dbe0103373fd758913933887761995510a51:

Everything seems all right :smile: .
*CFME QE Bot*

File: utils/providers.py [hunk 1]
```diff
@@ -762,16 +762,17 @@ def get_crud(provider_config_name):
...
                 prov_config['amqp_credentials'], cred_type='amqp')
+
         return OpenStackProvider(name=prov_config['name'],
             hostname=prov_config['hostname'],
             ip_address=prov_config['ipaddress'],
             api_port=prov_config['port'],
-            credentials={
-                'default': credentials,
-                'amqp': amqp_credentials},
+            credentials=custom_credentials,
```
Comment (psav) [inline utils/providers.py]: Hmm... if discovery uses the same "endpoints" system then maybe we need to update the discovery to loop the credentials? So discovery on openstack uses amqp?

Comment (jistefan) [inline utils/providers.py]: It probably does not, the problem here was that amqp_credentials was not mandatory and created only if present but referenced every time.

**Comment by dajoRH** [comment]
Lint report for commit a1754e6f071754598a0149928cbab4a69c917132:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 42773a5bfb9d93db9be172f5ee04210edd0acf34:

Everything seems all right :smile: .
*CFME QE Bot*

File: utils/providers.py (hunk 1 above)
Comment (jteehan) [inline utils/providers.py]: This is a popular subject this week.  Perhaps we should get openstack and azure done at the same time, and strip that out of 2977.

File: cfme/tests/infrastructure/test_discover_multiple_providers.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            continue
+        signatures_seen.add(sig)
+        start_ip, max_range = minmax_ip(prov_comb)
+        providers_final.append([prov_comb, start_ip, max_range])
+        new_id_list.append(sig)
+
+    testgen.parametrize(metafunc, argnames, providers_final, ids=new_id_list, scope="module")
+
+
+@pytest.yield_fixture(scope='module')
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_discover_multiple_providers.py]: I think this should be `"function"` scoped, anyway I think this is good to be merged.

**Comment by dajoRH** [comment]
Lint report for commit 47f0938272b225b3ebfe4619e15b833aff4335dc:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 984d4a03add9269aa23924cd533486f4ba1c609c:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request closed without merging**